src/core/thread/mle.cpp

```diff
@@ -569,7 +566,7 @@ otError Mle::BecomeChild(AttachMode aMode)
 
     netif.GetMeshForwarder().SetRxOnWhenIdle(true);
 
-    mParentRequestTimer.Start((otPlatRandomGet() % kParentRequestRouterTimeout) + 1);
+    mParentRequestTimer.Start(Random::GetUint32InRange(1, kParentRequestRouterTimeout + 1));
```

Comment:
nit: here you put the `+1` inside, whereas below you put the `+1` outside.  They are equivalent, but I prefer to be consistent.

Reply:
Looks like I missed some. Will change them. Thanks.
I initially used `+1` inside, but then switched to `+1` outside (to be (a bit) safer if the given max is close to int max limit when we use `uint8` or `uint16` versions of the random methods to avoid any int wrap around).